metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -167,7 +160,9 @@ class MetalsLanguageServer(
   val excludedPackageHandler: ExcludedPackagesHandler =
     new ExcludedPackagesHandler(userConfig.excludedPackages)
   var ammonite: Ammonite = _
-  val buildTargets: BuildTargets = BuildTargets.withAmmonite(() => ammonite)
+  private val mainBuildTargetsData = BuildTargets.Data.create()
```

Comment:
Why do we need to create empty data here? Could we not create it when indexing?

Reply:
This sounds like "why make things immutable when we could make them mutable?" to me 😅

Creating it upfront allows to pass it around and "wire" things upfront.

Reply:
Sure, I was just thinking that in some cases you don't actually need those targets. For example when there is only ammonite ones.